### PR TITLE
Python 3 support (for #176)

### PR DESCRIPTION
**Pull request by brianmartin**
This adds a tox env for testing against py35 (Beam is only supporting 3.5 for now).

### CODE REVIEW

**Comment by andrewsmartin** [comment]
@brianmartin now that beam 2.11 has been released as well as tfx 0.13, i think we can update this and see what breaks.

**Comment by brianmartin** [comment]
@andrewsmartin Cool, updated! I'll have a look tomorrow at what's changed that one of the py2 tests is failing and then look a bit at fixing things up for py3 (there's at least some str/bytes issues, etc.).

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,15 +22,25 @@ jobs:
...
-  test:
+  test-py27:
+    executor: tox
+    steps:
+      - tox: {env: "test-py27,upload-coverage"}
+
+  test-py37:
+    executor: tox
+    steps:
+      - tox: {env: "test-py27"}
```
Comment (andrewsmartin) [inline .circleci/config.yml]: `test-py27` -> `test-py37`?

File: README.md [hunk 2]
```diff
@@ -49,16 +49,18 @@ To see all `testenv`'s:
...
+test-py37
+upload-coverage
+license
 ```
 
 To run the tests:
 
 ```
-tox -e test
+tox -e test-py27
```
Comment (andrewsmartin) [inline README.md]: `tox -e test-py27,test-py37`?

File: requirements.txt [hunk 3]
```diff
@@ -3,8 +3,8 @@ numpy>=1.14.0,<2.0.0
 oauth2client<4.0.0
 pandas>=0.22.0,<1.0.0
 six>=1.11.0,<2.0.0
-tensorflow==1.13.1; python_version=='2.7'
-tensorflow-transform==0.12.0; python_version=='2.7'
-tensorflow-data-validation==0.12.0; python_version=='2.7'
+tensorflow==1.13.1
+tensorflow-transform==0.13.0
+tensorflow-data-validation==0.13.0
 tensorflow_metadata==0.12.1
```
Comment (andrewsmartin) [inline requirements.txt]: Let's bump this too? https://github.com/tensorflow/metadata/releases/tag/v0.13.0

File: requirements.txt [hunk 4]
```diff
@@ -3,8 +3,8 @@ numpy>=1.14.0,<2.0.0
 oauth2client<4.0.0
 pandas>=0.22.0,<1.0.0
 six>=1.11.0,<2.0.0
-tensorflow==1.13.1; python_version=='2.7'
-tensorflow-transform==0.12.0; python_version=='2.7'
-tensorflow-data-validation==0.12.0; python_version=='2.7'
+tensorflow==1.13.1
+tensorflow-transform==0.13.0
+tensorflow-data-validation==0.13.0
```
Comment (andrewsmartin) [inline requirements.txt]: `0.13.1` has  been released with a bugfix. https://github.com/tensorflow/data-validation/releases/tag/v0.13.1

**Comment by brianmartin** [comment]
Responded to @andrewsmartin's comments and fixed up all 2->3 test failures. There remains one test failure (on both py2 and 3) around representation of sparse vectors. This must be related to an upstream change.

**Comment by brianmartin** [comment]
So this is the detail on the only substantive change (at least as caught by our tests):

```
======================================================================
FAIL: test_example_with_feature_spec_decoder (tests.example_decoders_test.ExampleDecodersTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/brianm/src/github/spotify/spotify-tensorflow/tests/example_decoders_test.py", line 82, in test_example_with_feature_spec_decoder
    self.assertEqual(actual_json, expected_decoded)
AssertionError: {u'scalar_feature_3': 1.0, u'scalar_feature_2': 12, u'scalar_feature_1': 12, u'i [truncated]... != {u'scalar_feature_3': 1.0, u'scalar_feature_2': 12, u'scalar_feature_1': 12, u'2 [truncated]...
  {u'1d_vector_feature': [u'this is a ,text'],
   u'2d_vector_feature': [[1.0, 2.0], [3.0, 4.0]],
-  u'idx': [1, 4],
   u'scalar_feature_1': 12,
   u'scalar_feature_2': 12,
   u'scalar_feature_3': 1.0,
-  u'value': [12.0, 20.0],
+  u'sparse_feature': [[1, 4], [12.0, 20.0]],
   u'varlen_feature_1': [89.0],
   u'varlen_feature_2': [u'female']}
```

Curious for your thoughts on this one @andrewsmartin.

**Comment by brianmartin** [comment]
Also note that this may just be cosmetic. The only usage I can find of this Example -> Json decoder is in `tfr-read`, which is intended for spot inspection and debugging.

**Pull request merged**